packages/waku/src/router/create-pages.ts

```diff
@@ -162,27 +162,33 @@ export function createPages(
       staticPathSet.add([page.path, pathSpec]);
       const id = joinPath(page.path, 'page').replace(/^\//, '');
       registerStaticComponent(id, page.component);
-    } else if (page.render === 'static' && numSlugs > 0 && numWildcards === 0) {
+    } else if (page.render === 'static' && numSlugs > 0) {
       const staticPaths = (
         page as {
           staticPaths: string[] | string[][];
         }
       ).staticPaths.map((item) => (Array.isArray(item) ? item : [item]));
       for (const staticPath of staticPaths) {
-        if (staticPath.length !== numSlugs) {
+        if (staticPath.length !== numSlugs && numWildcards === 0) {
           throw new Error('staticPaths does not match with slug pattern');
         }
         const mapping: Record<string, string> = {};
         let slugIndex = 0;
-        const pathItems = pathSpec.map(({ type, name }) => {
-          if (type !== 'literal') {
-            const actualName = staticPath[slugIndex++]!;
-            if (name) {
-              mapping[name] = actualName;
-            }
-            return actualName;
+        const pathItems = [] as string[];
+        pathSpec.forEach(({ type, name }) => {
+          switch (type) {
+            case 'literal':
+              pathItems.push(name!);
+              break;
+            case 'wildcard':
+              staticPath.slice(slugIndex++).forEach((slug) => {
+                pathItems.push(slug);
+              });
+              break;
+            case 'group':
+              pathItems.push(staticPath[slugIndex++]!);
+              break;
```

Comment:
I did find my way in reasonable time, so it can't be that bad 😄 
I fully agree with the comment on line 29, but there is no unit test framework. I suppose we would use vitest?
 https://github.com/dai-shi/waku/blob/098b6b3f2c8a5af20919392f92259ad9bda2840f/packages/waku/src/router/create-pages.ts#L29

Happy to take a stab at this, in a separate PR.

Reply:
That would be more than helpful!
Yes, let's add vitest. A separate PR would be perfect.

Reply:
So the unit tests immediately revealed a regression! 😅

Reply:
That's nice! Yeah, because I was very unsure that I could review the code.